astroquery/vo/utils.py

```diff
@@ -0,0 +1,258 @@
+
+    return None
+
+#
+# Functions to help replace bytes with strings in astropy tables that came from VOTABLEs
+#
+
+def sval(val):
+    """
+    Returns a string value for the given object.  When the object is an instanceof bytes,
+    utf-8 decoding is used.
+
+    Parameters
+    ----------
+    val : object
+        The object to convert
+
+    Returns
+    -------
+    string
+        The input value converted (if needed) to a string
+    """
+    if isinstance(val, bytes):
+        return str(val, 'utf-8')
+    else:
+        return str(val)
+
+# Create a version of sval() that operates on a whole column.
+svalv = np.vectorize(sval)
+
+def sval_whole_column(single_column):
+    """
+    Returns a new column whose values are the string versions of the values
+    in the input column.  The new column also keeps the metadata from the input column.
+
+    Parameters
+    ----------
+    single_column : astropy.table.Column
+        The input column to stringify
+
+    Returns
+    -------
+    astropy.table.Column
+        Stringified version of input column
+    """
+    new_col = svalv(single_column)
+    new_col.meta = single_column.meta
+    return new_col
+
+def stringify_table(t):
+    """
+    Substitutes strings for bytes values in the given table.
+
+    Parameters
+    ----------
+    t : astropy.table.Table
+        An astropy table assumed to have been created from a VOTABLE.
+
+    Returns
+    -------
+    astropy.table.Table
+        The same table as input, but with bytes-valued cells replaced by strings.
+    """
+    # This mess will look for columns that should be strings and convert them.
+    if len(t) is 0:
+        return   # Nothing to convert
+
+    scols = []
+    for col in t.columns:
+        colobj = t.columns[col]
+        if (colobj.dtype == 'object' and isinstance(t[colobj.name][0], bytes)):
+            scols.append(colobj.name)
+
+    for colname in scols:
+        t[colname] = sval_whole_column(t[colname])
+
+def query_loop(query_function, service, params, verbose=False):
+    # Only one service, which is expected to be a row of a Registry query result that has  service['access_url']
+    if verbose: print("    Querying service {}".format(html.unescape(service['access_url'])))
+    # Initialize a table to add results to:
+    service_results = []
+    for j, param in enumerate(params):
+
+        result = query_function(service=html.unescape(service['access_url']), **param)
+        # Need a test that we got something back. Shouldn't error if not, just be empty
+        if verbose:
+            if len(result) > 0:
+                print("    Got {} results for parameters[{}]".format(len(result), j))
+                #Tracer()()
+            else:
+                print("    (Got no results for parameters[{}])".format(j))
+
+        service_results.append(result)
+    return service_results
+
+
+def try_query(url, retries=3, timeout=60, get_params=None, post_data=None, files=None):
```

Comment:
this function should be a method of the `Registry`.  Then instead of `bq._request`, you'll use `self._request`.

Reply:
This function is used in other classes Image, Cone, etc. and is not specific to Registry.

Reply:
I agree that this function belongs in the Registry class.  As we propose other classes that share this functionality, I think a small refactoring would be in order so that Registry and any new classes that want that function would inherit from a common superclass.

Reply:
Addressed:  The try_query function was moved into the (new) VoBase class, from which Registry now inherits.